text/0477-policy-validation.md

```diff
@@ -0,0 +1,548 @@
+violations to the user accordingly. This launch allows users to integrate one or more of these tools in the CDK 
+synthesis flow, so that, if their application produces a non-compliant CloudFormation template, they get the 
+feedback immediately.
+
+### Why should I use this feature?
+
+Your productivity as a CDK application developer will increase by getting quicker feedback on whether your 
+application complies with the policies defined for your whole organization. Instead of waiting for your deployment 
+pipeline to detect that some template is non-compliant, you can get this feedback immediately, every time you 
+synthesize the CDK application. If your application is not compliant, the CLI will report error messages, how to fix 
+the issues (depending on the integrated tool) and the offending construct.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+One of the recurring complaints about the CDK from large enterprise customers is that it generates many resources
+implicitly. While this is the original intent of the CDK, it also creates a challenge from a security and compliance
+standpoint; the generated resource definitions might not satisfy the rules defined by the central operations teams,
+which, at best, creates delays — as a result of development rework if the issues are caught before deployment — and, at
+worst, may expose the customer to attacks or lawsuits.
+
+By integrating policy-as-code tools in the synthesis flow, we give the confidence that these customers need to adopt the
+CDK at a larger scale.
+
+### What is the technical solution (design) of this feature?
+
+See [Appendix A - High level design](https://quip-amazon.com/lDPqAisXjVP6#temp:C:CaFafef0ee17e804c3bb00434ff5).
+
+### Is this a breaking change?
+
+No.
+
+### What alternative solutions did you consider?
+
+We have considered two alternative solutions:
+
+#### Validation in the CLI
+
+With this solution, the validation would be done in the CLI, rather than in the framework. With this approach, users
+would configure the validation plugins using a config file (cdk.json would be a natural candidate):
+
+```json
+"validations": {
+    "package": "cdk-cfn-guard",
+    "class": "CfnFuardValidation",
+    "version": "1.2.3"
+}
+```
+
+The CLI would read this file, load the appropriate plugins either from the filesystem or directly from NPM, and call
+them to get a validation report.
+
+We eventually discarded this option for the following reasons:
+
+* All plugins would have to be written in TypeScript/JavaScript for them to be consumable by the CLI.
+* The framework has access to the construct tree in memory, with all the contextual information that is needed to
+  provide a better error message, such as stack traces. This is useful to point out, for example, in which line 
+  an offending construct was defined. The CLI, on the other hand, only has access to the synthesized template.
```

Comment:
Not accurate, the CLI also has access to the `manifest.json` file, which can contain metadata produced by the framework. Its just easier to access this information from the framework, but possible via the CLI as well.

Reply:
Fair enough. Item removed.